tests/unit/test_wrroc_models.py

```diff
@@ -0,0 +1,235 @@
+        }
+        model = WRROCProcess(**data)
+        self.assertEqual(model.id, "process-id")
+        self.assertEqual(model.name, "Test Process")
+
+    def test_wrroc_process_empty_object_list(self):
+        """
+        Test that the WRROCProcess model handles empty object lists correctly.
+        """
+        data = {
+            "id": "process-id",
+            "name": "Test Process",
+            "object": []
+        }
+        model = WRROCProcess(**data)
+        self.assertEqual(model.object, [])
+
+    def test_wrroc_workflow_model(self):
+        """
+        Test that the WRROCWorkflow model correctly validates data and includes additional workflow fields.
+        """
+        data = {
+            "id": "workflow-id",
+            "name": "Test Workflow",
+            "workflowType": "CWL",
+            "workflowVersion": "v1.0",
+            "result": [{"id": "https://github.com/elixir-cloud-aai/CrateGen/blob/main/LICENSE", "name": "Output 1"}]
+        }
+        model = WRROCWorkflow(**data)
+        self.assertEqual(model.workflowType, "CWL")
+        self.assertEqual(model.result[0]['name'], "Output 1")
+
+    def test_wrroc_workflow_missing_optional_fields(self):
+        """
+        Test that the WRROCWorkflow model handles missing optional fields correctly.
+        """
+        data = {
+            "id": "workflow-id",
+            "name": "Test Workflow"
+        }
+        model = WRROCWorkflow(**data)
+        self.assertIsNone(model.workflowType)
+        self.assertIsNone(model.workflowVersion)
+
+    def test_wrroc_provenance_model(self):
+        """
+        Test that the WRROCProvenance model correctly validates data and includes additional provenance fields.
+        """
+        data = {
+            "id": "provenance-id",
+            "name": "Test Provenance",
+            "provenanceData": "Provenance information",
+            "agents": [{"id": "agent1", "name": "Agent 1"}]
+        }
+        model = WRROCProvenance(**data)
+        self.assertEqual(model.provenanceData, "Provenance information")
+        self.assertEqual(model.agents[0]['name'], "Agent 1")
+
+    def test_wrroc_provenance_empty_agents_list(self):
+        """
+        Test that the WRROCProvenance model handles empty agents lists correctly.
+        """
+        data = {
+            "id": "provenance-id",
+            "name": "Test Provenance",
+            "agents": []
+        }
+        model = WRROCProvenance(**data)
+        self.assertEqual(model.agents, [])
+
+    def test_wrroc_process_invalid_data(self):
+        """
+        Test that the WRROCProcess model raises a ValidationError with invalid data.
+        """
+        data = {
+            "id": 123,  # id should be a string
+            "name": None  # name should be a string
+        }
+        with self.assertRaises(ValidationError):
+            WRROCProcess(**data)
```

Comment:
1. Move up so that it follows the other `test_wrroc_process` tests.
2. Make sure that you also provide `_invalid_data` tests for `wrroc_workflow` and `wrroc_provenance` models (and put them in the right position).

Reply:
You could also consider defining separate classes for the different WRROC profile types, i.e., one for `process`, one for `workflow`, one for `provenance`.

Reply:
I’ve reorganized the test classes and ensured that each validator has its dedicated test class. Tests for invalid data are now correctly placed within their respective classes.